app/models/protocol.rb

```diff
@@ -478,28 +453,22 @@ def has_line_items_of_type?(current_request, portal, type)
     end
   end
 
-  def direct_cost_total service_request
-    total = 0
-    self.service_requests.each do |sr|
-      next if ['first_draft'].include?(sr.status) && sr != service_request
-      total += sr.direct_cost_total
-    end
-    return total
+  def direct_cost_total(service_request)
+    service_requests.where('status != ? OR id = ?', 'first_draft', service_request.id).
+      to_a.sum(&:direct_cost_total)
   end
 
-  def indirect_cost_total service_request
-    total = 0
+  def indirect_cost_total(service_request)
     if USE_INDIRECT_COST
-      self.service_requests.each do |sr|
-        next if ['first_draft', 'draft'].include?(sr.status) && sr != service_request
-        total += sr.indirect_cost_total
-      end
+      service_requests.where('(status != ? AND status != ?) OR id = ?', 'first_draft', 'draft', service_request.id).
```

Comment:
service_requests.where(id: service_request.id).where.not(status: ['first_draft', 'draft'])